docs/conf.py

```diff
@@ -361,3 +377,26 @@
 
 def setup(app):
     app.add_js_file("js/banner.js")
+
+    # Django-specific roles, from
+    # https://github.com/django/django/blob/main/docs/_ext/djangodocs.py:
+    app.add_crossref_type(
+        directivename="setting",
+        rolename="setting",
+        indextemplate="pair: %s; setting",
+    )
+    app.add_crossref_type(
+        directivename="templatetag",
+        rolename="ttag",
+        indextemplate="pair: %s; template tag",
+    )
+    app.add_crossref_type(
+        directivename="templatefilter",
+        rolename="tfilter",
+        indextemplate="pair: %s; template filter",
+    )
+    app.add_crossref_type(
+        directivename="fieldlookup",
+        rolename="lookup",
+        indextemplate="pair: %s; field lookup type",
+    )
```

Comment:
When using the `inv:` syntax in MyST, we don't need to add these to link to the roles. For example, we can already do `[](inv:django:std:setting#DATE_INPUT_FORMATS)`.

However, there's no such syntax in rST. I tried different ways but it seems that Sphinx won't pick up these custom roles. See: https://stackoverflow.com/questions/13387125

Adding this allows us to do `` :setting:`django:DATE_INPUT_FORMATS` `` in rST docs (docstrings and autodoc). It also allows us to use it in MyST like `` {setting}`django:DATE_INPUT_FORMATS` `` (or simply `` {setting}`DATE_INPUT_FORMATS` ``).